src/main/scala/chisel3/tester/internal/TestEnvInterface.scala

```diff
@@ -13,6 +18,24 @@ trait TestEnvInterface {
 
   def topFileName: Option[String]
 
+  /**
+    * Will add a TargetDirAnnotation with defaultDir with "test_run_dir" path prefix to the annotations
+    * if there is not a TargetDirAnnotation already present
+    *
+    * @param defaultDir     a default directory
+    * @param annotationSeq  annotations to add it to, unless one is already there
+    * @return
+    */
+  def addDefaultTargetDir(defaultDir: String, annotationSeq: AnnotationSeq): AnnotationSeq = {
+    if(annotationSeq.exists { case _: TargetDirAnnotation => true ; case _ => false }) {
```

Comment:
I think this may be a case where isInstanceOf produces cleaner code than the case matching.

Reply:
Uses `isInstanceOf` now